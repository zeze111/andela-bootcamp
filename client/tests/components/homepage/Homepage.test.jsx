import React from 'react';

import connectedHomepage, {Homepage} from '../../../components/homepage/Homepage';
import store from '../../../store';

const props = {
  getAllRecipes: jest.fn(() => Promise.resolve()),
  getMostUpvotedRecipe: jest.fn(() => Promise.resolve()),
  getPopularRecipes: jest.fn(() => Promise.resolve()),
  recipes: [
    {
      id: 1,
      name: 'something'
    },
    {
      id: 2,
      name: 'another thing'
    }
  ],
  upvotedRecipes: [
    {
      id: 1,
      name: 'something'
    },
    {
      id: 2,
      name: 'another thing'
    }
  ],
  popularRecipes: [
    {
      id: 1,
      name: 'something'
    },
    {
      id: 2,
      name: 'another thing'
    }
  ]
}

$ = () => {
  tooltip: () => {}
};

const setup = () => {
  const wrapper = shallow(<Homepage {...props} store={store} />);
  return { wrapper }
}

describe('Test for Homepage', () => {
  it('should render component', () => {
    const wrapper = setup();
    '<div>' +
    '  <Link class="button-collapse" />' +
    '  <div class="tooltipped" />' +
    '</div>';

    expect(wrapper).toMatchSnapshot()
  });
});
