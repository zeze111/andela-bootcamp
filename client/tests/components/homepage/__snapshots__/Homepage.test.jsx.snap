// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Homepage should render component 1`] = `
Object {
  "wrapper": <div
    id="homepageBody"
  >
    <main>
      <Component />
      <div
        className="container full-container home-width"
      >
        <br />
         
        <br />
        <h5
          className="light black-text text-headers lighten-3"
        >
          TRENDING RECIPES
        </h5>
        <div
          className="row"
        >
          <div
            className="center-align loader-style min-preloader"
          >
            <PreLoader />
          </div>
        </div>
      </div>
      <div
        className="container full-container home-width"
      >
        <br />
         
        <br />
        <h5
          className="light black-text text-headers lighten-3"
        >
          POPULAR RECIPES
        </h5>
        <div
          className="row"
        >
          <div
            className="center-align loader-style min-preloader"
          >
            <PreLoader />
          </div>
        </div>
      </div>
      <div
        className="container full-container home-width"
      >
        <br />
         
        <br />
        <Link
          href="/all-recipes"
          replace={false}
          to="/all-recipes"
        >
          <h5
            className="light text-color text-headers"
          >
             ALL RECIPES 
          </h5>
        </Link>
        <div
          className="row remove-margin-bottom"
        >
          <div
            className="center-align loader-style min-preloader"
          >
            <PreLoader />
          </div>
        </div>
      </div>
    </main>
  </div>,
}
`;
